src/world/StatusEffect/StatusEffect.cpp

```diff
@@ -85,33 +135,84 @@ uint16_t Sapphire::StatusEffect::StatusEffect::getParam() const
 void Sapphire::StatusEffect::StatusEffect::applyStatus()
 {
   m_startTime = Util::getTimeMs();
+
   auto pScriptMgr = m_pFw->get< Scripting::ScriptMgr >();
+  pScriptMgr->onStatusReceive( m_targetActor, m_id );
 
-  // this is only right when an action is being used by the player
-  // else you probably need to use an actorcontrol
-
-  //GamePacketNew< FFXIVIpcEffect > effectPacket( m_sourceActor->getId() );
-  //effectPacket.data().targetId = m_sourceActor->getId();
-  //effectPacket.data().actionAnimationId = 3;
-  //effectPacket.data().unknown_3 = 1;
-  //effectPacket.data().actionTextId = 3;
-  //effectPacket.data().unknown_5 = 1;
-  //effectPacket.data().unknown_6 = 321;
-  //effectPacket.data().rotation = ( uint16_t ) ( 0x8000 * ( ( m_sourceActor->getPos().getR() + 3.1415926 ) ) / 3.1415926 );
-  //effectPacket.data().effectTargetId = m_sourceActor->getId();
-  //effectPacket.data().effects[4].unknown_1 = 17;
-  //effectPacket.data().effects[4].bonusPercent = 30;
-  //effectPacket.data().effects[4].param1 = m_id;
-  //effectPacket.data().effects[4].unknown_5 = 0x80;
-  //m_sourceActor->sendToInRangeSet( effectPacket, true );
+  switch( static_cast< Common::StatusEffectType >( m_effectEntry.effectType ) )
+  {
+    case Common::StatusEffectType::Dot:
+    {
+      auto wepDmg = Sapphire::Math::CalcStats::getWeaponDamage( m_sourceActor );
+      auto damage = Sapphire::Math::CalcStats::calcDamageBaseOnPotency( *m_sourceActor, m_effectEntry.effectValue2, wepDmg );
 
-  pScriptMgr->onStatusReceive( m_targetActor, m_id );
+      for( auto const& entry : m_sourceActor->getStatusEffectMap() )
+      {
+        auto status = entry.second;
+        auto effectEntry = status->getEffectEntry();
+        if( static_cast< Common::StatusEffectType >( effectEntry.effectType ) != Common::StatusEffectType::DamageMultiplier )
+          continue;
+        if( effectEntry.effectValue1 & m_effectEntry.effectValue1 )
+        {
+          damage *= 1.0f + ( effectEntry.effectValue2 / 100.0f );
+        }
+      }
+
+      m_value = Sapphire::Math::CalcStats::applyDamageReceiveMultiplier( *m_targetActor, damage,
+        m_effectEntry.effectValue1 == static_cast< int32_t >( Common::ActionTypeFilter::Physical ) ? Common::AttackType::Physical :
+        ( m_effectEntry.effectValue1 == static_cast< int32_t >( Common::ActionTypeFilter::Magical ) ? Common::AttackType::Magical : Common::AttackType::Unknown_0 ) );
+      m_cachedSourceCrit = Sapphire::Math::CalcStats::criticalHitProbability( *m_sourceActor, Common::CritDHBonusFilter::Damage );
+      m_cachedSourceCritBonus = Sapphire::Math::CalcStats::criticalHitBonus( *m_sourceActor );
+      break;
+    }
+
+    case Common::StatusEffectType::Hot:
+    {
+      auto wepDmg = Sapphire::Math::CalcStats::getWeaponDamage( m_sourceActor );
+      auto heal = Sapphire::Math::CalcStats::calcHealBaseOnPotency( *m_sourceActor, m_effectEntry.effectValue2, wepDmg );
```

Comment:
why are these calculations duplicated even though they're used in the exact same way in both handlers?